lib/logstash/outputs/scalyr.rb

```diff
@@ -343,7 +347,15 @@ def multi_receive(events)
             @logger.error(message, exc_data)
             exc_commonly_retried = false
           end
-          retry if @running
+          retry if @running and exc_retries < @max_retries
+          message = "Failed to send event after #{exc_retries} tries."
```

Comment:
Should we include a truncated sample of some events in the batch in the log message to make troubleshooting easier?

Reply:
And perhaps also include error message from the last retry in this log / error message.

Reply:
The last error message is actually included, we log the `exc_data` object which includes the message among other things.